js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,137 @@
+            expect(enhancedDate.getDate(true)).to.be.an.instanceOf(Date);
+        });
+    });
+    
+    
+    //	function getDayName() {
+    describe('getDayName', function () {
+        
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day), so this is May 31, 2015
+            enhancedDate.setDate(new Date(2015, 4, 31));
+        });
+        
+        it("should return a string", function () {
+            expect(enhancedDate.getDayName()).to.be.a('string');
+        });
+        
+        it("should return 'Sunday'", function () {
+            expect(enhancedDate.getDayName()).to.equal('Sunday');
+        });
+    });
+        
+    //	function getMonthName() {
+    describe("getMonthName", function () {
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day) so this is June 2, 2015
+            enhancedDate.setDate(new Date(2015, 5, 2));
+        });
+        
+        it("should return a string", function () {
+            expect(enhancedDate.getMonthName()).to.be.a('string');
+        });
+        
+        it("should return 'June'", function () {
+            expect(enhancedDate.getMonthName()).to.equal('June');
+        });
+        
+
+    });
+    
+    //	function isToday() {
+    describe("isToday", function () {
+            
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day) so this is June 2, 2015
+            // set a past date so it will never equal today
+            enhancedDate.setDate(new Date(2015, 5, 2));
+        });
+        
+        it("should return a boolean", function () {
+            expect(enhancedDate.isToday()).to.be.a('boolean');
+        });
+        
+        it("should return not equals today", function () {
+            expect(enhancedDate.isToday()).to.not.equal(new Date());
+        });
+        
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day) so this is June 2, 2015
+            // set a past date so it will never equal today
+            enhancedDate.setDate(new Date());
+        });
+        
+        it("should return equals today", function () {
+            expect(enhancedDate.isToday()).to.equal(true);
+        });
+
+    });
+    
+//	function isFuture() {
+    describe("isFuture", function () {
+            
+        before (function () {
+            // remember: Date (year, month # starting with Jan == 0, day) so this is June 2, 3000
+            // set a future date so it will return true
+            enhancedDate.setDate(new Date(3000, 5, 2));
+        });
+    
+        it("should return a boolean", function () {
+            expect(enhancedDate.isFuture()).to.be.a('boolean');
+        });
+        
+        it("should return false, it is not the future", function () {
+            expect(enhancedDate.isFuture()).to.not.equal(new Date());
```

Comment:
Same thing as with isToday. In this case, passing new Date(), while it causes the test to pass, uses false logic to do so. Instead, this would be more clear: `expect(enhancedDate.isFuture90).to.be.false;`